src/main/java/net/minecraftforge/fml/common/registry/GameRegistry.java

```diff
@@ -210,12 +205,39 @@ public static Block registerBlock(Block block, Class<? extends ItemBlock> itemcl
                 Constructor<? extends ItemBlock> itemCtor = itemclass.getConstructor(ctorArgClasses);
                 i = itemCtor.newInstance(ObjectArrays.concat(block, itemCtorArgs));
             }
+            return registerBlock(block, i, name);
+        }
+        catch (Exception e)
+        {
+            FMLLog.log(Level.ERROR, e, "Caught an exception during block registration");
+            throw new LoaderException(e);
+        }
+    }
+    
+    /**
+     * Register a block with the world, with the specified item and block name. <br />
+     * This method will register both the block and the item
+     * @param block The block to register
+     * @param item The item to register with it : null registers a block without associated item.
+     * @param name The mod-unique name to register it as, will get prefixed by your modid.
+     * 
+     */
+    public static Block registerBlock(Block block, ItemBlock item, String name)
+    {
+        if (Loader.instance().isInState(LoaderState.CONSTRUCTING))
+        {
+            FMLLog.warning("The mod %s is attempting to register a block whilst it it being constructed. This is bad modding practice - please use a proper mod lifecycle event.", Loader.instance().activeModContainer());
+        }
+        try
+        {
+            assert block != null : "registerBlock: block cannot be null";
```

Comment:
Maybe replace this with a normal null check, throwing a nice exception if the block is null?